### DISCUSSION TITLE
At what moment flag FORCED WRITES is set during backup restore?

### DISCUSSION THREAD

**Discussion by gsbelarus** [FirebirdSQL/firebird#7944] [Ideas]
There is a significant performance difference between FORCED WRITES ON and OFF modes. Would it be great to restore database with the flag set to OFF and only upon successful finish set it to ON or OFF according to the previous value from the backup or to the parameter, specified during restoring?

**Comment by AlexPeshkoff** [comment]
On 1/2/24 20:07, Andrej Kirejeŭ wrote:
>
> There is a significant performance difference between FORCED WRITES ON 
> and OFF modes. Would it be great to restore database with the flag set 
> to OFF and only upon successful finish set it to ON or OFF according 
> to the previous value from the backup or to the parameter, specified 
> during restoring?
>

You've exactly described how it works currently ;)

**Comment by gsbelarus** [comment]
Bad for me. I need to get rid of habits from the nineties.
